.changeset/nasty-drinks-judge.md

```diff
@@ -0,0 +1,27 @@
+---
+'@makeswift/runtime': minor
+---
+
+This change updates the runtime to use the latest version of
+`@makeswift/controls`. As part of this update, this package is no longer
+exposing internal data types and functions associated with our controls.
+
+## BREAKING:
+
+1. Registering controls with arbitrary options will now result in a TypeScript
+   error. For example, suppose you currently had the following registration:
+
+   ```typescript
+   import { Number } from '@makeswift/runtime/controls'
+
+   const num = Number({ foo: 'bar' })
+   ```
+
+   `foo` is not a valid `Number` control option, resulting in an error. Any
+   invalid options in your component registrations must be removed for
+   successful compilation.
+
+2. When passing `options` to a `Select` control, you must provide at least one
```

Comment:
2. `Select` control now requires the `options` array to have at least one entry:
    
   ```typescript
    import { Select } from '@makeswift/runtime/controls'

    const sel = Select({
      label: 'Block type',
      options: [] // error, non-empty array is required
    })
   ```

    Previously, the `options` parameter was allowed to be empty.